feat: add network request example with layered architecture

bogy0 opened the pull request:
This PR adds a simple end-to-end example of how to fetch data. The goal is to give developers a real example how to do this.
**(FYI: This PR is a rework of #87. The original branch was created about three months ago and diverged significantly from main, so I recreated the feature on a fresh branch based on the latest main.)**

This PR also gives example how to access params/queries in client component

### What’s included
#### PostComponent example
Added a small React component that loads a "blog post" + its "comments". It shows:
- how to call the API,
- and how to keep the UI clean by not mixing fetch logic into the component.

### Clear separation between layers
- UI layer: React component that just renders stuff
`src/pages/welcome/post/PostComponent.jsx`
- API layer: tiny client-side wrappers around the backend routes
`src/api/message.js`
- Service layer: server-side handlers that talk to an external API (mocked here)
`src/service/postHandlers.js, src/service/externalHandlers.js`

This should make the overall flow easier to follow and copy/paste for future features, also testing can be easier.

### New endpoints
- GET /api/post/:id
- GET /api/comments?postId=:id
- GET /api/external/post/:id
- GET /api/external/comments?postId=:id

These external endpoints are mocked 
— they’re only here to demonstrate how the service layer would normally proxy to some real third-party service.

### Dashboard extended with handling url params/queries
- `/dashboard/1234?q=xxx&name=Anne`

#### This is how this new section looks in the welcome page:
<img width="3310" height="1924" alt="Screenshot 2025-11-20 at 14 25 38" src="https://github.com/user-attachments/assets/ff3eb4db-909e-4cfa-bc6b-5e15c5a9bee5" />

#### Modified dashboard page, to demonstrate how to handle params/queries client-side:
<img width="3362" height="3274" alt="Screenshot 2025-11-20 at 15 48 14" src="https://github.com/user-attachments/assets/2ed9e7e7-91de-4289-ab1d-d2dbf72cf9fb" />

bogy0 committed:
feat(api): implement external API mock handlers and update routing

- Added externalHandlers.js to simulate external API responses for posts and comments.
- Updated routes to include new external API mock routes.
- Refactored postHandlers to call the new external API endpoints.
- Adjusted server.js to export base URL for easier configuration.
- Enhanced test setup to mock external API calls correctly.

Resolves: no-ticket
Signed-off-by: [Balint Lendvai] <lendvai.balint@gmail.com>

bogy0 committed:
refactor(server): extract server configuration to a separate module

- Moved server configuration constants (port and base) from server.js to a new config/server-config.js file for better organization and testability.
- Updated imports in postHandlers.js and test files to reflect the new configuration module.
- Adjusted response handling in tests to use HttpResponse from msw.

Resolves: no-ticket

bogy0 committed:
feat(nav, dashboard): enhance navigation and dashboard with dynamic routing and parameter handling

- Added a utility function to determine active menu items based on current path, supporting dynamic routes.
- Updated NavHeaderItems to utilize the new active path logic.
- Enhanced Dashboard component to access and display URL path and query parameters.
- Configured routing to support dynamic dashboard paths with parameters.

Resolves: no-ticket

bogy0 committed:
fix: query param typo fix

rodet commented on src/test/server.js:
```diff
@@ -14,9 +15,43 @@ const _setupServer = (...args) => {
+        userId: 1,
+      });
+    }),
+    http.get('http://localhost:5173/api/external/comments', ({ request }) => {
```
Shouldn't this use to $PORT env variable?

rodet commented on src/pages/welcome/Welcome.jsx:
```diff
@@ -154,14 +142,38 @@ const Welcome = () => {
+                    state and renders the data using Carbon Design components
+                  </ListItem>
+                  <ListItem>
+                    <strong>API Layer</strong> - Client-side functions in{' '}
```
Casing

rodet commented on src/pages/dashboard/Dashboard.jsx:
```diff
@@ -40,6 +50,46 @@ const Dashboard = () => {
+              <Stack gap={3}>
+                {id && (
+                  <div>
+                    <strong>Path Parameter (id):</strong>{' '}
```
Consider using those markup tags instead
```
<dl>
  <dt>Name</dt>
  <dd>John Doe</dd>
  <dt>Email</dt>
  <dd>john.doe@example.com</dd>
</dl>
```

rodet reviewed (dismissed):
Looks good overall! Please review my comments. Later on we should add more test cases but this could be a separate PR.

bogy0 committed:
fix: sentence case fixes and some pollishing

Resolves: no-ticket

bogy0 committed:
fix: update post title in externalHandlers

lee-chase committed:
feat: fix preview:prod

lee-chase reviewed (commented):
@bogy0 made a small number of changes to fix preview:prod.

Turns out that at run time Express insists on the whole object being selected.

I was also a bit confused about the posts/comments.

The id and postId always seemed to be one. 
In addition it would be nice to prevent invalid IDs. Suggest 400 for invalid ID, 404 for out of range/not found.

bogy0 committed:
feat: enhance PostComponent to accept dynamic postId prop

also added JSDoc for PostComponent

Resolves: no-ticket

bogy0 commented:
> Hi @bogy0, perhaps my understanding of what we are demonstrating is incorrect. I expected postId and id to match some back end identifier and thought by changing the params I might see

@lee-chase now i understand, sorry for my confusion, so what i did to address this in my latest commit (https://github.com/carbon-design-system/carbon-react-router-starter/pull/279/commits/9a3fb40f6f3ef90b63903190caa2c8cf93389954) is that i added a `postId` property to the `<PostComponent />` and also added  JSDoc, so at the Welcome page now its controllable which post we want to load.
